### PR TITLE
Pre-commit-ci workflow

### LINKED ISSUE #135: Recommend users pre-commit.ci
pre-commit.ci is a CI service running pre-commit hooks - and committing fixes to open PRs. It also keeps `.pre-commit-config.yaml` updated sending new PRs to bump individual hook versions. pre-commit.ci requires no additional configuration - just `.pre-commit-config.yaml`. I suggest updating docs to recommend it for users.

### PR DESCRIPTION
**Pull request by camposandro**
Implements a workflow to integrate with [pre-commit.ci lite](https://pre-commit.ci/lite). The goal is to configure a bot that performs automatic code styling fixes on open pull requests, with hooks that:

* Clear outputs from Jupyter notebooks
* Sort imports using ``isort`` (python files)
* According to the preferred linter selected:
   * Format code using ``black`` (python and notebook files)
   * Check compliance with ``pylint`` rules (python files)

Closes #135.

## Checklist

- [X] This PR is meant for the `lincc-frameworks/python-project-template` repo and not a downstream one instead.
- [X] This change is linked to an open issue
- [X] This change includes integration testing, or is small enough to be covered by existing tests

### CODE REVIEW

File: python-project-template/.github/workflows/{% if preferred_linter != 'none' %}pre-commit-ci.yml{% endif %}.jinja [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+name: Run pre-commit hooks
+
+on:
+  pull_request:
+
+jobs:
+  pre-commit-ci:
+    runs-on: ubuntu-latest
+    env: 
+      SKIP: "check-lincc-frameworks-template-version,pytest-check,no-commit-to-branch,validate-pyproject,check-added-large-files,sphinx-build"
```
Comment (camposandro) [inline python-project-template/.github/workflows/{% if preferred_linter != 'none' %}pre-commit-ci.yml{% endif %}.jinja:14]: I don't like having this huge skip environment variable here. Any ideas on how I could improve this?

Comment (delucchi-cmu) [reply python-project-template/.github/workflows/{% if preferred_linter != 'none' %}pre-commit-ci.yml{% endif %}.jinja:14]: Is the concern just that it's concatenating all the names together, and looks gross?

I agree that it looks gross, but it's probably ok. I'm not sure if the behavior of `env:` is to set or append. If append, you could split it up like (including the commas at the end of the lines):

```
env:
  SKIP: "check-lincc-frameworks-template-version,"
  SKIP: "pytest-check,"
  SKIP: "no-commit-to-branch,"
  SKIP: "validate-pyproject,"
  ...
```

Or you can try some github action language operators like:

```
env:
  SKIP: join(["check-lincc-frameworks-template-version",
                  "pytest-check",
                  "no-commit-to-branch",
                  "validate-pyproject",
                  "check-added-large-files",
                  "sphinx-build"], ',')
```

Comment (drewoldag) [reply python-project-template/.github/workflows/{% if preferred_linter != 'none' %}pre-commit-ci.yml{% endif %}.jinja:14]: Either of @delucchi-cmu suggestions look good to me, but I would be ok with just leaving it as is. It's not so long that a user couldn't read through it and pretty much understand what's going on.

File: python-project-template/.github/workflows/{% if preferred_linter != 'none' %}pre-commit-ci.yml{% endif %}.jinja [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        sudo apt-get update
+        python -m pip install --upgrade pip
+        pip install .
+        pip install .[dev]
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+    - uses: pre-commit/action@v3.0.0
+      with:
+        extra_args: --from-ref {% raw %}${{ github.event.pull_request.base.sha }}{% endraw %} --to-ref {% raw %}${{ github.event.pull_request.head.sha }}{% endraw %}
+    - uses: pre-commit-ci/lite-action@v1.0.1
+      if: always()
```
Comment (drewoldag) [inline python-project-template/.github/workflows/{% if preferred_linter != 'none' %}pre-commit-ci.yml{% endif %}.jinja:34]: Is the `if: always()` necessary here? I don't know enough to know.

File: python-project-template/.pre-commit-config.yaml.jinja [hunk 3]
```diff
@@ -101,17 +104,18 @@ repos:
 {% elif preferred_linter == 'black' %}
     # Analyze the code style and report code that doesn't adhere.
   - repo: https://github.com/psf/black
-    rev: 23.1.0
+    rev: 23.7.0
     hooks:
-      - id: black
+      - id: black-jupyter
```
Comment (drewoldag) [inline python-project-template/.pre-commit-config.yaml.jinja:108]: Why `black-jupyter` here?

**Review by drewoldag — APPROVED** [review]
This looks pretty good to me. I left a couple comments that would be nice to get resolved prior to merging. But overall, this is great!

File: python-project-template/.github/workflows/{% if preferred_linter != 'none' %}pre-commit-ci.yml{% endif %}.jinja (hunk 2 above)
Comment (camposandro) [reply python-project-template/.github/workflows/{% if preferred_linter != 'none' %}pre-commit-ci.yml{% endif %}.jinja:34]: I've followed this [recommendation](https://pre-commit.ci/lite) from the **pre-commit.ci lite** webpage.

File: python-project-template/.pre-commit-config.yaml.jinja (hunk 3 above)
Comment (camposandro) [reply python-project-template/.pre-commit-config.yaml.jinja:108]: This change allows _black_ to format Jupyter notebooks.

File: python-project-template/.github/workflows/{% if preferred_linter != 'none' %}pre-commit-ci.yml{% endif %}.jinja (hunk 1 above)
Comment (camposandro) [reply python-project-template/.github/workflows/{% if preferred_linter != 'none' %}pre-commit-ci.yml{% endif %}.jinja:14]: Alright, if it's no hassle I left as is!

**Pull request merged**